### PR TITLE
[CL]: Deprecate duplicated queries from gamm module

### PR DESCRIPTION
**Pull request by mattverse**
Closes: https://github.com/osmosis-labs/osmosis/issues/3883

## What is the purpose of the change

This PR deprecates three of the following queries from the gamm module since they have been moved to the swap rotuer module. 
- NumPools
- EstimateSwapExactAmountIn
- EstimateSwapExactAmountOut

Protos remained undeleted but marked deprecated using the deprecation option in proto3.

This PR also removes the three queries from the stargate whitelist list. 

## Brief Changelog

- Deprecates duplicated queries between the gamm module and the swap router module.

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? yes
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? yes
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -67,6 +67,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * [#3817](https://github.com/osmosis-labs/osmosis/pull/3817) Move osmoassert from `app/apptesting/osmoassert` to `osmoutils/osmoassert`.
 * [#3771](https://github.com/osmosis-labs/osmosis/pull/3771) Move osmomath into its own go.mod
 * [#3827](https://github.com/osmosis-labs/osmosis/pull/3827) Move osmoutils into its own go.mod
+* [#3827](https://github.com/osmosis-labs/osmosis/pull/3907) Add swaprouter queries `Numpools`, `EstimateSwapExactAmountIn` and `EstimateSwapExactAmountOut` to stargate whitelist.
```
Comment (p0mvn) [inline CHANGELOG.md:70]: I think this changelog entry is referring to another PR
```suggestion
* [#3905](https://github.com/osmosis-labs/osmosis/pull/3905) Deprecate gamm queries `NumPools`, `EstimateSwapExactAmountIn` and `EstimateSwapExactAmountOut`.
```

File: wasmbinding/stargate_whitelist.go [hunk 2]
```diff
@@ -73,7 +73,6 @@ func init() {
 	setWhitelistedQuery("/osmosis.epochs.v1beta1.Query/CurrentEpoch", &epochtypes.QueryCurrentEpochResponse{})
 
 	// gamm
-	setWhitelistedQuery("/osmosis.gamm.v1beta1.Query/NumPools", &gammtypes.QueryNumPoolsResponse{})
```
Comment (p0mvn) [inline wasmbinding/stargate_whitelist.go:76]: Let's not remove these please to avoid breaking old contracts

**Review by p0mvn — COMMENTED** [review]
LGTM once comments are addressed and CI passes

File: wasmbinding/stargate_whitelist.go (hunk 2 above)
Comment (mattverse) [reply wasmbinding/stargate_whitelist.go]: Brought them back for now, same question here, do we have plans to deprecate them in the future? I think V14 upgrade would be a good point to introduce this API break wdyt?

Comment (p0mvn) [reply wasmbinding/stargate_whitelist.go]: We discussed this in another PR thread so I'm going to resolve this one

File: tests/e2e/configurer/chain/commands.go [hunk 3]
```diff
@@ -25,12 +25,12 @@ func (n *NodeConfig) CreatePool(poolFile, from string) uint64 {
 	_, _, err := n.containerManager.ExecTxCmd(n.t, n.chainId, n.Name, cmd)
 	require.NoError(n.t, err)
 
-	path := "osmosis/gamm/v1beta1/num_pools"
+	path := "osmosis/swaprouter/v1beta1/num_pools"
```
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:28]: We should use the deprecated path in e2e, at least for now.

The reason is that e2e runs before the upgrade and after the upgrade.

So before upgrade, there are no swaprouter queries existing yet. As a result, e2e fails

File: tests/e2e/configurer/chain/commands.go [hunk 4]
```diff
@@ -25,12 +25,12 @@ func (n *NodeConfig) CreatePool(poolFile, from string) uint64 {
...
 
 	bz, err := n.QueryGRPCGateway(path)
 	require.NoError(n.t, err)
 
-	var numPools gammtypes.QueryNumPoolsResponse
+	var numPools swaprouterqueryproto.NumPoolsResponse
```
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:33]:
```suggestion
	var numPools gammtypes.NumPoolsResponse
```

File: tests/e2e/configurer/chain/commands.go (hunk 3 above)
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:28]:
```suggestion
	path := "osmosis/gamm/v1beta1/num_pools"
```

File: tests/e2e/configurer/chain/commands.go [hunk 5]
```diff
@@ -12,8 +12,8 @@ import (
 	appparams "github.com/osmosis-labs/osmosis/v13/app/params"
 	"github.com/osmosis-labs/osmosis/v13/tests/e2e/configurer/config"
 	"github.com/osmosis-labs/osmosis/v13/tests/e2e/util"
-	gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
 	lockuptypes "github.com/osmosis-labs/osmosis/v13/x/lockup/types"
+	swaprouterqueryproto "github.com/osmosis-labs/osmosis/v13/x/swaprouter/client/queryproto"
```
Comment (p0mvn) [inline tests/e2e/configurer/chain/commands.go:16]:
```suggestion
        gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
	lockuptypes "github.com/osmosis-labs/osmosis/v13/x/lockup/types"
```

**Review by p0mvn — COMMENTED** [review]
LGTM once CI passes

**Review by czarcas7ic — APPROVED** [review]
It seems you still need to remove the three queries from the stargate whitelist

After than and Roman's comments this LGTM

**Review by czarcas7ic — COMMENTED** [review]
After reading on of the other PRs, it looks like we need to keep these on the stargate whitelist. Probably just need to add a linter staticcheck

File: tests/e2e/configurer/chain/commands.go (hunk 3 above)
Comment (mattverse) [reply tests/e2e/configurer/chain/commands.go:28]: makes sense to me

File: tests/e2e/configurer/chain/commands.go (hunk 4 above)
Comment (mattverse) [reply tests/e2e/configurer/chain/commands.go:33]: fixed

**Pull request merged**